truffle/com.oracle.truffle.api/src/com/oracle/truffle/api/impl/Accessor.java

```diff
@@ -163,6 +163,16 @@ protected Object eval(TruffleLanguage<?> l, Source s, Map<Source, CallTarget> ca
         return API.eval(l, s, cache);
     }
 
+    /**
+     * @param vm
+     * @param ev
+     * @param code text to evaluate
+     * @param node node in halted context
+     * @param frame frame in halted context
+     * @return result of language evaluation
+     * @throws IOException
+     * @throws UnsupportedOperationException if not supported by the language implementation
+     */
     protected Object evalInContext(Object vm, Object ev, String code, Node node, MaterializedFrame frame) throws IOException {
```

Comment:
I think I have an idea how to clean up the accessor pattern a bit and reduce the number of indirections. If you keep remind me, I'll do it at the end. What @kamosh thinks about that?